### ISSUE TITLE
Support for React 17

### ISSUE THREAD

**Issue by no23reason** [xiel/use-location-state#265]
Hi, I was wondering if there are any plans to officially support React 17 (in the `peerDependencies` field of the packages), please?
It would prevent warnings on installation along the lines of
```
warning " > react-router-use-location-state@3.0.0" has incorrect peer dependency "react@^16.8.0".
```
in applications using React 17.

The package already seems to be working fine with React 17, so I guess this would only be a matter of expanding the `peerDependencies`.

Thanks

**Comment by xiel** [comment]
Thanks for opening this issue, I published a new version with updated peerDependencies.

Fixed in v3.0.1 📦

**Comment by no23reason** [comment]
Hi @xiel thank you for your changes :) however, looking at the [commit](https://github.com/xiel/use-location-state/commit/ebe27305cb63d503ef3ab97b198ee0dd814c5c32#), it seems that this is a breaking change (now React 16 is not supported). I suggest you change the `peerDependencies` to `"react": "^16.8.0 || ^17.0.2"` so as to keep the compatibility the version 3.0.0 had. I opened a PR fixing this: #273 please consider it.
